var/spack/repos/builtin/packages/python/package.py

```diff
@@ -454,10 +463,12 @@ def write_easy_install_pth(self, exts):
                             re.search(r'setuptools.*egg$', line)):
                         continue
 
-                    paths.append(line)
+                    line = os.path.realpath(line)
```

Comment:
Some of the easy-install.pth files contain lines like:

```./Babel-2.3.4-py2.7.egg```

In which case realpath may expand relative to the current directory (vs. the directory where the easy-install.pth file is)

As you mention, it appears that https://github.com/LLNL/spack/pull/3587 will do away with this. I'll add a comment there that perhaps this entire function should disappear. If it can't for some reason then IMO this issue should be fixed.